.travis.yml

```diff
@@ -79,6 +79,12 @@ jobs:
         skip_cleanup: true
         on:
           branch: master
+      - provider: script
+        script: npm config set //registry.npmjs.org/:_authToken=$NPM_PUBLISH_TOKEN && node utils/publish-canary.js $SLACK_WEBHOOK $TRAVIS_BUILD_WEB_URL prerelease
+        skip_cleanup: true
+        on:
+          all_branches: true
+          condition: $TRAVIS_BRANCH =~ ^prerelease\/v\d*$
```

Comment:
Not sure if it's possible to put an inline conditional in the script line to only add the `prerelease` preid if the branch matches the regex. You also need to keep the script tag on one line for deploys. Seemed easier to just duplicate the master canary script.

Travis docs are bad 🤷‍♂️ - if anyone knows of a better way to do this, let me know.

Reply:
Would writing a script and leveraging Travis env variables accomplish this? I'm not sure if they're available node context but you could always do something like: `TRAVIS_BRANCH=$TRAVIS_BRANCH node canary-script.js`

And then somewhere in the node script:
```
if (process.env.TRAVIS_BRANCH.match(regex) !== null) // etc
```

Reply:
I feel like all Travis scripts inevitably end up devolving into a file somewhere that gets checked in...

Reply:
@lychyi Just pushed an update - let me know what you think